kinto_http/session.py

```diff
@@ -79,14 +79,22 @@ def request(self, method, endpoint, data=None, permissions=None,
                         retry_after = self.retry_after
                     time.sleep(retry_after)
                     continue
+            else:
```

Comment:
I don't think you have to do a `else` here, since `Backoff` can be set on any status code. You could even move above the `retry = retry - 1`